### PR TITLE
BPMSPL-252: [GSS-RFE] Ability to check duplicate of ReleaseId (GroupId, ArtifactId and Version) with existing projects. Updates following QE feedback.

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/BPMSPL-252

This PR includes updates requested by QE following their testing. It also moves a couple of WebBeans (WAS) unfriendly ```<? extends Project>``` from ```guvnor``` to ```<KieProject>``` in ```kie-wb-common```.

Branch name retained (yes, I know @psiroky it's not ideal) to be consistent and support reconciliation of all changes made for the feature.

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:-

https://github.com/droolsjbpm/guvnor/pull/252
https://github.com/droolsjbpm/kie-wb-common/pull/263

File: guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/test/java/org/guvnor/asset/management/client/editors/repository/wizard/CreateRepositoryWizardTest.java [hunk 1]
```diff
@@ -569,6 +573,230 @@ public void testManagedRepositoryMultiCompletedClashingGAVForcedTest() {
     }
 
     @Test
+    @SuppressWarnings("unchecked")
+    public void testManagedRepositoryMultiCompletedClashingGAVBeforeRepositoryInitializationTest() {
```
Comment (triceo) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/test/java/org/guvnor/asset/management/client/editors/repository/wizard/CreateRepositoryWizardTest.java]: Thank you, @manstis. :-)
(Seriously - thanks for the tests. This insane, yet correct, method name is just the cherry on top.)

Comment (manstis) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/test/java/org/guvnor/asset/management/client/editors/repository/wizard/CreateRepositoryWizardTest.java]: @triceo Glad you like it - when I start to need to break method names over 2 lines I think I'll need a new strategy!

**Comment by Rikkola** [comment]
Tried to catch you. "I'm sure he did not test failing service calls". But you did ;-)
+1

**Pull request merged**